workers/unity/Packages/com.improbable.gdk.core/Utility/EntityBuilder.cs

```diff
@@ -0,0 +1,215 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Improbable.Worker.Core;
+
+namespace Improbable.Gdk.Core
+{
+    public class EntityBuilder
+    {
+        private readonly Entity entity;
+        private Acl acl = new Acl();
+        private bool hasBuiltOnce;
+
+        private readonly HashSet<uint> componentsAdded = new HashSet<uint>();
+
+        public static EntityBuilder Begin()
+        {
+            return new EntityBuilder();
+        }
+
+        protected EntityBuilder()
+        {
+            entity = new Entity();
+        }
+
+        public EntityBuilder AddComponent(ComponentData componentData, string writeAccess)
+        {
+            if (componentsAdded.Contains(componentData.ComponentId))
+            {
+                throw new InvalidOperationException(
+                    $"Cannot add multiple components of the same type to the same entity. " +
+                    $"Attempted to add componentId: {componentData.ComponentId} more than once.");
+            }
+
+            entity.Add(componentData);
+            componentsAdded.Add(componentData.ComponentId);
+            acl.SetComponentWriteAccess(componentData.ComponentId, writeAccess);
+            return this;
+        }
+
+        public EntityBuilder AddPosition(double x, double y, double z, string writeAccess)
+        {
+            var schemaData = new SchemaComponentData(WellKnownComponents.Position.ComponentId);
+            var fields = schemaData.GetFields();
+            fields.AddDouble(1, x);
+            fields.AddDouble(2, y);
+            fields.AddDouble(3, z);
+
+            return AddComponent(new ComponentData(schemaData), writeAccess);
+        }
+
+        public EntityBuilder SetPersistence(bool persistence)
+        {
+            if (persistence)
+            {
+                var schemaData = new SchemaComponentData(WellKnownComponents.Persistence.ComponentId);
+                entity.Add(new ComponentData(schemaData));
+            }
+            else
+            {
+                if (entity.Get(WellKnownComponents.Persistence.ComponentId).HasValue)
```

Comment:
While magic numbers are bad there is a lot of indirection going on here. I think you can make the magic numbers const locals and remove all the well known component stuff. It doesn't really make things easier to have a WellKnownComponents class.